File: tests/wpt/web-platform-tests/scroll-to-text-fragment/stash.js
// Put test results into Stash
function stashResults(key, results) {
  fetch(`/scroll-to-text-fragment/stash.py?key=${key}`, {
    method: 'POST',
    body: JSON.stringify(results)
  });
}

// Fetch test results from the Stash
function fetchResults(key, resolve, reject) {
  fetch(`/scroll-to-text-fragment/stash.py?key=${key}`).then(response => {
    return response.text();
  }).then(text => {
    if (text) {
      try {
        const results = JSON.parse(text);
        resolve(results);
      } catch(e) {
        reject();
      }
    }
  });
}

File: tests/wpt/web-platform-tests/scroll-to-text-fragment/stash.py
import time

def main(request, response):
    key = request.GET.first("key")

    if request.method == "POST":
        # Received result data from target page
        request.server.stash.put(key, request.body, '/scroll-to-text-fragment/')
        return "ok"
    else:
        # Request for result data from test page
        value = request.server.stash.take(key, '/scroll-to-text-fragment/')
        return value
